play-json/shared/src/main/scala/JsMacroImpl.scala

```diff
@@ -25,16 +25,16 @@ import scala.reflect.macros.blackbox
     }
   }
 
-  def formatImpl[A: c.WeakTypeTag]: c.Expr[OFormat[A]] =
-    macroImpl[A, OFormat, Format](
+  def formatImpl[A: c.WeakTypeTag, O <: Json.MacroOptions]: c.Expr[OFormat[A]] =
```

Comment:
One problem with this is that now there are two different ways to pass `JsonConfiguration` (using `Json.configure`, or just defining an implicit `JsonConfiguration` in the macros scope), however, only one of them allows passing macro options.  So if I do this:

```scala
implicit val myConfiguration = JsonConfiguration[WithDefaultValues](SnakeCase)
implicit val fooReads: Reads[Foo] = Json.reads[Foo]
```

I will find the surprising behavior that my reads will use snake case, however they won't use default values, because `Json.reads` is hard coded to use the default options. Is it not possible to extract the options out of the implicitly passed JsonConfiguration type member?